components/sidebar/sidebar_whats_new_modal/sidebar_whats_new_modal.tsx

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {Preferences} from 'mattermost-redux/constants';
+import {PreferenceType} from 'mattermost-redux/types/preferences';
+
+import GenericModal from 'components/generic_modal';
+
+import './sidebar_whats_new_modal.scss';
+
+type Props = {
+    currentUserId: string;
+    hasSeenModal: boolean;
+    newSidebarPreference: boolean;
+    actions: {
+        savePreferences: (userId: string, preferences: PreferenceType[]) => void;
+    };
+}
+
+type State = {
+    isUsingNewSidebar: boolean;
+}
+
+export default class SidebarWhatsNewModal extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            isUsingNewSidebar: props.newSidebarPreference,
+        };
+    }
+
+    onHide = () => {
+        this.props.actions.savePreferences(this.props.currentUserId, [{
+            user_id: this.props.currentUserId,
+            category: Preferences.CATEGORY_WHATS_NEW_MODAL,
+            name: Preferences.HAS_SEEN_SIDEBAR_WHATS_NEW_MODAL,
+            value: 'true',
+        }]);
+    }
+
+    render() {
+        return (
+            <GenericModal
+                show={!this.props.hasSeenModal && this.state.isUsingNewSidebar}
+                onHide={this.onHide}
+                modalHeaderText={(
+                    <FormattedMessage
+                        id={'sidebar_whats_new_modal.header'}
+                        defaultMessage={'What\'s new'}
+                    />
+                )}
+                handleConfirm={() => {}}
```

Comment:
This should also be omitted if possible.

Reply:
Can we make this prop optional?